src/fparser/two/tests/fortran2003/test_include_statement.py

```diff
@@ -0,0 +1,109 @@
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+'''Test Fortran Include Statement: This file tests the parsing of an
+include statement. Whilst include is not part of the standard Fortran
+rules (the include should include code as the code is being parsed)
+there are cases where users might like to keep the include statement
+in the Fortran parse tree and output it again.
+
+'''
+
+import pytest
+from fparser.api import get_reader
+from fparser.two.Fortran2003 import Include_Stmt
+from fparser.two.utils import NoMatchError
+
+
+def test_include_stmt(f2003_create):
+    '''Check that a basic include statement is parsed
+    correctly. Input separately as a string and as a reader object
+
+    '''
+    def check_include(reader):
+        '''Internal helper function to avoid code replication.'''
+        ast = Include_Stmt(reader)
+        assert "INCLUDE 'my-non-existant-file.inc'" in str(ast)
+        assert repr(ast) == ("Include_Stmt(Include_Filename("
+                             "'my-non-existant-file.inc'))")
+
+    line = "include 'my-non-existant-file.inc'"
+    check_include(line)
+    reader = get_reader(line)
+    check_include(reader)
+
+
+def test_spaces(f2003_create):
+    '''Check that spaces are allowed before and after an include keyword
+    as well as after the file string.
+
+    '''
+    line = " include 'my-non-existant-file.inc' "
+    ast = Include_Stmt(line)
+    assert "INCLUDE 'my-non-existant-file.inc'" in str(ast)
+
+
+def test_no_space(f2003_create):
+    '''Check that no space is required between the include keyword and the
+    file string.
+
+    '''
+    line = "include'my-non-existant-file.inc'"
+    ast = Include_Stmt(line)
+    assert "INCLUDE 'my-non-existant-file.inc'" in str(ast)
+
+
+def test_case(f2003_create):
+    '''Check that different case is allowed for the include keyword.'''
+    line = "InClUdE 'my-non-existant-file.inc'"
+    ast = Include_Stmt(line)
+    assert "INCLUDE 'my-non-existant-file.inc'" in str(ast)
+
+
+def test_double_quotes(f2003_create):
+    '''Check that double quotes are allowed for the file string.'''
+    line = 'include "my-non-existant-file.inc"'
+    ast = Include_Stmt(line)
+    assert "INCLUDE 'my-non-existant-file.inc'" in str(ast)
+
+
+def test_errors(f2003_create):
+    '''Check that syntax errors produce a NoMatchError exception.'''
+    for line in [None, "", "  ", "includ", "include", "includ 'x'", "include",
```

Comment:
I think you have "include" twice in this list.

Reply:
So I did. Well spotted.